modules/caddyhttp/matchers.go

```diff
@@ -513,7 +513,16 @@ func (m *MatchHeader) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 
 // Match returns true if r matches m.
 func (m MatchHeader) Match(r *http.Request) bool {
-	return matchHeaders(r.Header, http.Header(m), r.Host)
+	repl := r.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
+	header := http.Header(m)
+	for field, allowedFieldVals := range header {
+		for index, allowedFieldVal := range allowedFieldVals {
+			allowedFieldVals[index] = repl.ReplaceAll(allowedFieldVal, "")
```

Comment:
We should probably only use `ReplaceAllKnown()` here so that it's still possible to match header values that have words surrounded in braces for some reason.

Reply:
Sounds good. I Changed it.